R/froll.R

```diff
@@ -1,21 +1,107 @@
-froll = function(fun, x, n, fill=NA, algo=c("fast", "exact"), align=c("right", "left", "center"), na.rm=FALSE, hasNA=NA, adaptive=FALSE) {
-  stopifnot(!missing(fun), is.character(fun), length(fun)==1L, !is.na(fun))
-  algo = match.arg(algo)
+partial2adaptive = function(x, n, align, adaptive) {
+  if (align=="center")
+    stopf("'partial' cannot be used together with align='center'")
+  if (is.list(x) && length(unique(vapply(x, length, 0L)))!=1L)
+    stopf("'partial' does not support variable length of columns in 'x'")
+  if (!(is.numeric(n) || (is.list(n) && all(vapply(n, is.numeric, FALSE)))))
+    stopf("n must be integer vector or list of integer vectors")
+  len = if (is.list(x)) length(x[[1L]]) else length(x)
+  verbose = getOption("datatable.verbose")
+  if (!adaptive) {
+    n = as.list(n) ## test 6006.032
```

Comment:
Maybe its just me, but I would expect that
`frollmean(1:6/2, 3:4, partial=TRUE)`
and
`frollmean(1:6/2, list(3:4), partial=TRUE)`
return the same. Maybe add the `if (!is.list(n)) n = list(n)` and move before if/else branching?

Reply:
why return the same?
having an error here is to remove ambiguous use of `n` arg.
For adaptive=FALSE argument `n` never needs to be a list because each window is a integer scalar, therefore integer vector is sufficiently enough to represent the all possible inputs.
```r
frollmean(1:6/2, list(1:6, c(3:4,3:4,3:4)), adaptive=TRUE)
```

Reply:
Also strange that `frollmean(1:6/2, 3:4, partial=FALSE)` works, `frollmean(1:6/2, list(3:4), partial=FALSE)` does not, telling that `list is accepted for adaptive TRUE` but `frollmean(1:6/2, list(3:4), partial=TRUE)` also works

Reply:
must have been fixed in next branches...
```r
frollmean(1:6/2, list(3:4), partial=TRUE)
#Error: n must be integer, list is accepted for adaptive TRUE
```

Reply:
> why return the same? having an error here is to remove ambiguous use of `n` arg. For adaptive=FALSE argument `n` **never** needs to be a list because each window is a integer scalar, therefore integer vector is sufficiently enough to represent the all possible inputs. On the other hand, adaptive=TRUE, list is necessary to represent vectorized `n`:

I see, ty for the explanation.